smac/optimizer/ei_optimization.py

```diff
@@ -675,6 +675,111 @@ def _maximize(
         return next_configs_by_acq_value
 
 
+class LocalAndSortedPriorRandomSearch(AcquisitionFunctionMaximizer):
+    """Implements SMAC's default acquisition function optimization.
+
+    This optimizer performs local search from the previous best points
+    according, to the acquisition function, uses the acquisition function to
+    sort randomly sampled configurations. Random configurations are
+    interleaved by the main SMAC code.
+
+    Parameters
+    ----------
+    acquisition_function : ~smac.optimizer.acquisition.AbstractAcquisitionFunction
+
+    config_space : ~smac.configspace.ConfigurationSpace
+
+    rng : np.random.RandomState or int, optional
+
+    max_steps: int
+        [LocalSearch] Maximum number of steps that the local search will perform
+
+    n_steps_plateau_walk: int
+        [LocalSearch] number of steps during a plateau walk before local search terminates
+
+    n_sls_iterations: int
+        [Local Search] number of local search iterations
+    """
+
+    def __init__(
+        self,
+        acquisition_function: AbstractAcquisitionFunction,
+        config_space: ConfigurationSpace,
+        uniform_config_space: ConfigurationSpace,
+        rng: Union[bool, np.random.RandomState] = None,
+        max_steps: Optional[int] = None,
+        n_steps_plateau_walk: int = 10,
+        n_sls_iterations: int = 10,
+        prior_sampling_fraction = 0.5
+    ):
+        super().__init__(acquisition_function, config_space, rng)
+        self.prior_random_search = RandomSearch(
+            acquisition_function=acquisition_function,
+            config_space=config_space,
+            rng=rng
+        )
+        self.uniform_random_search = RandomSearch(
+            acquisition_function=acquisition_function,
```

Comment:
If I understand your question correctly, you're asking about PiBO as a whole. (?)

The key idea is to multiply the acquisition function by the prior, so PriorAcq = Acq * Prior (to the power of a factor < 1 so that the prior decays over time).

However, it becomes necessary to ensure the local search is not too biased by the prior, as it will sample from the configspace when it conducts the local search. As such, we have two configspaces.

Reply:
Thanks for your clarification, I was wondering about the sampling, and/but I got it now